gui/test/app/Stanc/Linting.test.ts

```diff
@@ -0,0 +1,112 @@
+import { assert, describe, expect, test } from "vitest";
+import { exportedForTesting } from "../../../src/app/Stanc/Linting";
+
+const { locationFromMessage, getWarningMessage, getErrorMessage } =
+  exportedForTesting;
+
+// all messages actually generated by stanc
+const jacobianWarning = `Warning in 'jacobian.stan', line 1, column 12: Variable name 'jacobian' will
+    be a reserved word starting in Stan 2.38.0. Please rename it!
+`;
+
+const multiColumnError = `Semantic error in 'multicol.stan', line 3, column 4 to column 10:
+   -------------------------------------------------
+     1:  parameters {
+     2:      real y;
+     3:      int x;
+             ^
+     4:  }
+     5:  model {
+   -------------------------------------------------
+
+(Transformed) Parameters cannot be integers.
+`;
+
+const multiLineError = `Semantic error in 'multiline.stan', line 2, column 2 to line 4, column 16:
+   -------------------------------------------------
+     1:  generated quantities {
+     2:    array[3] int x
+           ^
+     3:             =
+     4:               10;
+   -------------------------------------------------
+
+Ill-typed arguments supplied to assignment operator =:
+The left hand side has type
+  array[] int
+and the right hand side has type
+  int
+`;
+
+describe("Linting", () => {
+  describe("position detection", () => {
+    test("random message returns undefined", () => {
```

Comment:
Consider adding a case where message is empty; the `if (!message)` line doesn't seem to be tested.